### ISSUE TITLE
Typescript and other packages that are distributed in transpiled form only

### ISSUE THREAD

**Issue by silkentrance** [lerna/lerna#1487]
I have here a use case that requires packages originally authored using typescript, to be transpiled and then distributed in their transpiled form only.

During development, I also do not want to depend upon the original code but rather on its transpiled form. For me, this serves as an initial integration test for the typings that are included with the transpiled package.

As such, I have set up a gulp based SCM process where on invocation of the distdev task, all packages in the monorepo will be transpiled and their output is written to a target/distdev folder, along with a stripped down and augmented package.json that only includes the required dependencies and no devDependencies whatsoever.

For this, I have been developing a custom, gulp task/plugin based framework for dealing with the monorepo, quite similar to lerna. But alas, keeping this in sync across multiple project is becoming a PITA now and I am looking into using lerna for managing my monorepos.

ATM though, lerna only supports linking of local dependencies to the root of the respective packages and not some link target beneath that root.

## Expected Behavior

lerna, based upon a global configuration option, and possibly also based upon a per package override, will link the package to either the package's location or the package's location augmented by a user specified relative path that points to the transpiled package.

## Current Behavior

lerna, upon creating the symbolic links to local dependencies will link to the package's root directory only.

## Possible Solution

I would expect lerna to allow me to define the link target either globally, e.g. via a --link-target option, which itself represents a relative path to the package's location.
Upon creating the symbolic links, lerna would then augment the package's location by that link target.

I have prepared an initial PR #1488 for this. Please have a look and tell me what you think and whether this is worthwhile of pursuing.

## Context

See the summary at the top.

## Your Environment

Standard MAC, latest node/npm, lerna from master.

**Comment by evocateur** [comment]
#91 #901 #1282

TL;DR: Point your leaf package.json [`main`](https://docs.npmjs.com/files/package.json#main) field to the transpiled output, and bob's your uncle.

**Comment by evocateur** [comment]
This is how npm packages have worked for a decade. Lerna is just managing multiple packages in a single repository.
